incidents/src/test/java/fr/openent/incidents/service/impl/IncidentsTypeServiceTest.java

```diff
@@ -0,0 +1,142 @@
+        vertx = Vertx.vertx();
+        Sql.getInstance().init(vertx.eventBus(), "fr.openent.incidents");
+        this.incidentsService = new DefaultIncidentsTypeService();
+    }
+
+    @Test
+    public void testFetchIncidentsType(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "SELECT * FROM null.incident_type where structure_id = ?";
+        JsonArray expectedParams = new JsonArray().add("structureId");
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "fetchIncidentsType", "structureId", (Handler) e -> {
+
+        });
+    }
+
+    @Test
+    public void testFetchUsedIncidentsType(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "WITH ids AS (SELECT i.id, i.label FROM null.incident_type i WHERE structure_id = ?) " +
+                "SELECT DISTINCT i.id, i.label FROM ids i WHERE (i.id IN (SELECT type_id FROM null.incident))";
+        JsonArray expectedParams = new JsonArray().add("structureId");
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", (Handler) e -> {
+
+        });
+    }
+
+    @Test
+    public void testCreate(TestContext ctx) throws Exception {
+        Async async = ctx.async();
+
+        String expectedQuery = "INSERT INTO null.incident_type (structure_id, label, hidden) VALUES (?, ?, false) RETURNING id";
+        JsonArray expectedParams = new JsonArray().add("structureId").add("label");
+        JsonObject incidentTypeBody = new JsonObject().put("structureId", "structureId").put("label", "label");
+
+        vertx.eventBus().consumer("fr.openent.incidents", message -> {
+            JsonObject body = (JsonObject) message.body();
+            ctx.assertEquals("prepared", body.getString("action"));
+            ctx.assertEquals(expectedQuery, body.getString("statement"));
+            ctx.assertEquals(expectedParams.toString(), body.getJsonArray("values").toString());
+            async.complete();
+        });
+
+        Whitebox.invokeMethod(this.incidentsService, "create", incidentTypeBody, (Handler) e -> {
+
```

Comment:
Whitebox.invokeMethod(this.incidentsService, "fetchUsedIncidentsType", "structureId", null)